### PR TITLE
ICMSLST-2440 Add views to allow a user to edit personal data in ICMS. 

### PR DESCRIPTION
**Pull request by MattHolmes123**
Changes:
  - Add UserUpdateView.
  - Add UserCreateTelephoneView.
  - Add UserUpdateTelephoneView.
  - Add UserDeleteTelephoneView.
  - Add UserCreateEmailView.
  - Add UserUpdateEmailView.
  - Add UserDeleteEmailView.
  - Add UserDetailView for an ILB Admin to view readonly user data.
  - Replace links to current-user-details to use user-edit view instead.
  - Add basic permission tests for each new view.

### CODE REVIEW

File: web/domains/user/urls.py [hunk 1]
```diff
@@ -1,10 +1,77 @@
...
+                            ),
+                        ]
+                    ),
+                ),
+            ]
+        ),
+    ),
+    path("", views.current_user_details, name="current-user-details"),
+    # Admin Views to view users in ICMS.
+    path("users/", views.UsersListView.as_view(), name="users-list"),
```
Comment (MattHolmes123) [inline web/domains/user/urls.py:75]: This old view will be deleted in the next ticket (ICMSLST-2436)

File: web/domains/user/urls.py [hunk 2]
```diff
@@ -1,10 +1,77 @@
...
+                                    ]
+                                ),
+                            ),
+                        ]
+                    ),
+                ),
+            ]
+        ),
+    ),
+    path("", views.current_user_details, name="current-user-details"),
```
Comment (MattHolmes123) [inline web/domains/user/urls.py:73]: This view will be deleted in the next ticket. ICMSLST-2436

**Comment by MattHolmes123** [comment]
image
image
image
image
image
image
image
image

**Comment by MattHolmes123** [comment]
Readyonly ILB Admin view of a users data (Accessed via Admin -> Web User Accounts)
image

**Review by mikeylockwood — APPROVED** [review]

**Pull request merged**